protocol/messenger_linkpreview.go

```diff
@@ -127,27 +128,44 @@ func GetURLs(text string) []string {
 }
 
 func NewDefaultHTTPClient() *http.Client {
-	return &http.Client{Timeout: unfurlers.DefaultRequestTimeout}
+	return &http.Client{Timeout: DefaultRequestTimeout}
 }
 
 // UnfurlURLs assumes clients pass URLs verbatim that were validated and
 // processed by GetURLs.
-func (m *Messenger) UnfurlURLs(httpClient *http.Client, urls []string) ([]common.LinkPreview, error) {
+func (m *Messenger) UnfurlURLs(httpClient *http.Client, urls []string) (UnfurlURLsResponse, error) {
 	if httpClient == nil {
 		httpClient = NewDefaultHTTPClient()
 	}
 
-	previews := make([]common.LinkPreview, 0, len(urls))
+	r := UnfurlURLsResponse{
+		LinkPreviews:       make([]common.LinkPreview, 0, len(urls)),
+		StatusLinkPreviews: make([]common.StatusLinkPreview, 0, len(urls)),
+	}
 
 	for _, url := range urls {
 		m.logger.Debug("unfurling", zap.String("url", url))
+
+		if m.IsStatusSharedURL(url) {
+			unfurler := NewStatusUnfurler(url, m, m.logger)
+			preview, err := unfurler.Unfurl()
+			if err != nil {
+				m.logger.Warn("failed to unfurl status link", zap.String("url", url), zap.Error(err))
+				continue
+			}
+			r.StatusLinkPreviews = append(r.StatusLinkPreviews, preview)
+			continue
+		}
+
 		p, err := m.unfurlURL(httpClient, url)
 		if err != nil {
-			m.logger.Info("failed to unfurl", zap.String("url", url), zap.Error(err))
+			m.logger.Warn("failed to unfurl", zap.String("url", url), zap.Error(err))
 			continue
 		}
-		previews = append(previews, p)
+		r.LinkPreviews = append(r.LinkPreviews, p)
 	}
 
-	return previews, nil
+	m.logger.Info("<<< UnfurlURLs", zap.Any("response", r))
```

Comment:
The response can be extremely large for logging since there can be a lot of image data encoded. I might be wrong, but I think we should be careful, maybe remove this or log a subset of the response without the encoded data URIs. In my past experiments I only temporarily logged responses due to their size.

Reply:
forgot to remove 👌